app/controllers/users_controller.rb

```diff
@@ -4,10 +4,10 @@ def create
     @user = User.new(user_params)
 
     if @user.save
-      redirect_to root_path, notice: 'User successfully created.'
       session[:user_id] = @user.id
+      redirect_to root_path, notice: 'User successfully created.'
     else
-      render :new
+      render :new, notice: 'Error: Unable to create user account'
```

Comment:
I think because of the way `render` works, you might need to set the flash separately:
```
flash[:error] = "Error: Unable to create user account"
```
That is, unless this works now in Rails 5.  I'd make sure that you're including a test for this messaging to make sure it's showing up when you'd expect.